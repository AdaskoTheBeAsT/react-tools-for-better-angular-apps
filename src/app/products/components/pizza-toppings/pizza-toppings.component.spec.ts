import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { PizzaToppingsComponent } from './pizza-toppings.component';
import { Pizza, Topping } from '../../models';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

describe(`${PizzaToppingsComponent.name}`, () => {
  let component: PizzaToppingsComponent;
  let fixture: ComponentFixture<PizzaToppingsComponent>;
  const toppingsMock: Topping[] = [
    {
      id: 1,
      name: 'anchovy',
    },
    {
      id: 2,
      name: 'bacon',
    },
  ];
  const pizzaMock: Pizza = {
    id: 1,
    name: 'Pepperoni pizza',
    toppings: [],
  };

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        declarations: [PizzaToppingsComponent],
        imports: [NoopAnimationsModule],
      }).compileComponents();
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(PizzaToppingsComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it(`should display pizza toppings`, () => {
    component.toppings = toppingsMock;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
