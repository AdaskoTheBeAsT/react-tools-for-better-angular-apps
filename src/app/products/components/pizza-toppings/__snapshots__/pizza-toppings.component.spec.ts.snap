// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PizzaToppingsComponent should display pizza toppings 1`] = `
<pizza-toppings
  toppings={[Function Array]}
  value={[Function Array]}
>
  
      <div
    class="pizza-toppings"
  >
    
        
    
    
      
  </div>
    
</pizza-toppings>
`;
